kate_brownbill_dungon_fighters_v1.0.py

```diff
@@ -0,0 +1,838 @@
+            
+def jester_game():
+    player = Player(0 ,"JESTER", "IT'S ME, IT'S YOU")
+    him = Fighter([FIREBOLT, FRWD_SWNG, MAGIC_SWNG, SHIELD_BASH, MAGIC_RFCT], "The Demonic Joker", "A sub-speaces of imp who is arguably the most powerful being in existence", 6, 6, 6)
+    lv = get_menu(["1", "2", "3", "4"], "Pick a dificalty level.\n[1] [2] [3] [4]\n", "That's not a valid levle.")
+    foe = Player(4, "THE ENEMY", "BORING")
+    team = []
+    for i in range(lv + 1):
+        black_moves = []
+        red_moves = []
+        while black_moves == red_moves:
+            black_moves = create_moves(4)
+            red_moves = create_moves(4)      
+        card_black = Fighter([MAGIC_RFCT, FRWD_SWNG, SHIELD_BASH, MAGIC_SWNG], "Dark Card", "Woven from shadow. Does it even exist?", 3 + i, 3 + i, 1 + i)
+        card_red = Fighter([MAGIC_RFCT, FIREBOLT, MAGIC_SWNG, SHIELD_BASH], "Red Card", "Woven from blood. Does it even exist?", 3 + i, 4 + 1, 0 + 1)
+        team.append(card_black)
+        team.append(card_red)
+    do_battle([him], team, player, foe)
+            
+menu = True
+jester = False           
+while menu == True:
+    if jester == False:
+        ans = get_menu(["dungeon crawl", "tutorial", "exit", "jevil"], "Pick a game mode\n[DUNGEON CRAWL] [TUTORIAL] [EXIT]\n", "Please pick a valid mode")
+    else:
+        ans = get_menu(["dungeon crawl", "tutorial", "exit", "jevil", "jester's game"], "Pick a game mode\n[DUNGEON CRAWL] [TUTORIAL] [JESTER'S GAME] [EXIT]\n", "Please pick a valid mode")
+    if ans == 0:
+        result = do_dungeon_crawl()            
+        if result == 1:
+            jester = True
+    elif ans == 1:
+        print("\nWelcome to dungeon fighters!")
+        print("Here you command a party of adventurers who fight various foes.")
+        print("There are 5 moves, FIREBOLT, FORWARDS SWING, MAGIC SWING")
+        print("SHIELD BASH, and MAGIC REFLECT. Every fighter has some")
+        print("combanation of moves. Once the two fighters have picked a move")
+        print("to use, the two fight. Depending on what moves they used, one")
+        print("fighter will gain the advantage in the battle and deal damage.")
+        print("\nEach fighter as 3 stats: HP ATTC DEF. HP is the hit points")
+        print("ATTC is linked directly to damage given by a fighter. DEF absorbs")
+        print("damage given by an oponent, however DEF will not compleatly absorb")
+        print("all the damage.\n")
+    elif ans == 2:
+        menu = False
+    elif ans == 3:
+        jester = True
+        print("Jester's game unlocked")
+    elif ans == 4:
+        jester_game()
+        
+
+#---------------------------UNITTESTING-------------------------
+def test_ai():
+    fighter1 = Fighter([FIREBOLT, MAGIC_SWNG], "fighter1", "a fighter", 1, 1, 1)
+    fighter2 = Fighter([SHIELD_BASH, FRWD_SWNG, MAGIC_SWNG], "fighter2", "a fighter", 1, 1, 1)
+    
+    turnA1 = agressive_ai(SHIELD_BASH, fighter1) #should return magic swing
```

Comment:
Often, when people write unit tests, they'll use concepts like actual results and golden results.  i.e.,

```
actual_turn1 = aggressive_ai( SHIELD_BASH, fighter1 )
golden_turn1 = (1, MIGHT_SWNG, fighter1 )
equal_assert( actual_turn1,  golden_turn1 )
```

Another command pattern is to test as you go...  i.e.

`equal_assert( agressive_ai( SHIELD_BASH, fighter1 ), (1, MAGIC_SWNG, fighter1 ))`